streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -640,6 +644,61 @@ public void shouldRespectCommitNeeded() {
         assertFalse(task.commitNeeded());
     }
 
+    @Test
+    public void shouldRestorePartitionTimeAfterRestartWithEosDisabled() {
+        task = createStatelessTask(createConfig(false));
+        task.initializeStateStores();
+        task.initializeTopology();
+
+        task.addRecords(partition1, singletonList(getConsumerRecord(partition1, DEFAULT_TIMESTAMP)));
+
+        task.process();
+
+        // timestamp will be committed here
+        task.commit();
+        assertTrue(task.decodeTimestamp(task.consumer.committed(partition1).metadata())
+                   == DEFAULT_TIMESTAMP);
+        // reset times here to artificially represent a restart
+        task.resetTimes();
+        assertTrue(task.partitionTime(partition1) == RecordQueue.UNKNOWN);
+
+        // timestamp would be updated here
+        task.initializeTaskTime();
+        assertTrue(task.partitionTime(partition1) == DEFAULT_TIMESTAMP);
+        assertTrue(task.streamTime() == DEFAULT_TIMESTAMP);
+    }
+
+    @Test
+    public void shouldRestorePartitionTimeAfterRestartWithEosEnabled() {
+        task = createStatelessTask(createConfig(true));
+        task.initializeStateStores();
+        task.initializeTopology();
+
+        task.addRecords(partition1, singletonList(getConsumerRecord(partition1, DEFAULT_TIMESTAMP)));
+
+        task.process();
+        // time stamp will be committed here
+        task.commit();
+        
+        //since consumer is mock, there is no real broker
+        //so we need to manually commit the information to stimulate broker
+        final Map<TopicPartition, OffsetAndMetadata> offsetMap = new HashMap<>();
+        final String encryptedMetadata = task.encodeTimestamp(DEFAULT_TIMESTAMP);
+        offsetMap.put(partition1, new OffsetAndMetadata(DEFAULT_TIMESTAMP, encryptedMetadata));
+        task.consumer.commitSync(offsetMap);
```

Comment:
This test setup defeats the purpose fo this test. If EOS is enabled, the producer is used to commit offsets, and thus, we should check if the producer does commit the corresponding metadata correctly.

Therefore, we need to change the test setup a little bit. In `createStatelessTask()`, we create an anonymous `ProducerSupplier` and we need to get hold off the generated mock-producer instance. We can then use `MockProducer#consumerGroupOffsetsHistory` to get the committed offsets and metadata.